extensions/assistedinject/src/com/google/inject/assistedinject/FactoryProvider2.java

```diff
@@ -363,7 +364,7 @@ public Method getFactoryMethod() {
       ImmutableMap.Builder<Method, MethodHandle> methodHandleBuilder = ImmutableMap.builder();
       for (Map.Entry<String, Method> entry : defaultMethods.entries()) {
         Method defaultMethod = entry.getValue();
-        MethodHandle handle = createMethodHandle(defaultMethod, factory);
+        MethodHandle handle = superMethodHandle(defaultMethod, factory);
         if (handle != null) {
           methodHandleBuilder.put(defaultMethod, handle);
         } else {
```

Comment:
we should add a TODO that when we drop jdk8 support we can rm this 'else' branch, because its only needed as a fallback since using reflection wasn't a reliable way to do things.